src/server/field.rs

```diff
@@ -62,13 +51,11 @@ where R: BufRead, F: FnOnce(&[StrHeader]) -> Ret {
         let buf = try!(r.fill_buf());
 
         if attempts == MAX_ATTEMPTS {
-            error!("Could not read field headers.");
-            // RFC: return an actual error instead?
-            return Ok(closure(&[]));
+            return Err(ParseHeaderError::Other("Unknown error while parsing headers".to_string()));
```

Comment:
So this would only be hit if `fill_buf()` returned too small a buffer too many times, it's kind of relying on `BoundaryReader::read_to_boundary()` getting more data each attempt but that's outside of convention for `BufRead` (all standard impls either return all their data at once or don't read more until the buffer is completely consumed). This loop should only need to repeat once, if at all. So it's not really an error, the stream is just behaving weirdly and we don't want to loop forever because that could be used to DoS the server running this.